core/src/gl/texture.cpp

```diff
@@ -42,13 +35,10 @@ Texture::~Texture() {
     }
 }
 
-bool Texture::loadImageFromMemory(const std::vector<char>& _data) {
+bool Texture::loadImageFromMemory(const uint8_t* data, size_t length) {
     unsigned char* pixels = nullptr;
-    int width, height, comp;
-
-    if (_data.size() != 0) {
-        pixels = stbi_load_from_memory(reinterpret_cast<const stbi_uc*>(_data.data()), _data.size(), &width, &height, &comp, STBI_rgb_alpha);
-    }
+    int width = 0, height = 0, comp = 0;
+    pixels = stbi_load_from_memory(data, length, &width, &height, &comp, STBI_rgb_alpha);
```

Comment:
Can we not skip this call if the length passed is 0?

Reply:
I feel it's not worth adding a branch here for that case, since it should not occur in normal use.